src/metabase/driver/sql/query_processor.clj

```diff
@@ -301,7 +312,8 @@
 
 (defmethod ->honeysql [:sql :joined-field]
   [driver [_ alias field]]
-  (binding [*table-alias* alias]
+  (binding [*table-alias*   alias
```

Comment:
This is ugly and it's making me sad. I'm torn between using a flag and introducing `*join-alias*`. In the latter case `->honeysql` for `:joined-field` would only set `*join-alias*`, and we'd use `(or *joined-alias* *source-alias)` where relevant.

Adjacent to the issues around `:source_alias` I mention above.